### PR TITLE
PIVOT-5271 Load bookmarks from jar

### PR DESCRIPTION
**Pull request by roxanedelpeyratactiveviam**

### CODE REVIEW

File: src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java [hunk 1]
```diff
@@ -109,44 +92,49 @@ private static File getSubDirectory(final String folderName) {
...
+          parent.putChild(childName.replace(Paths.JSON, ""), node, true);
+        } else {
+          final SnapshotContentTree childNode = createEmptyDirectoryNode();
+          SnapshotContentTree childTree =
+              createDirectoryTree(root + Paths.SEPARATOR + childName, childNode);
+          parent.putChild(childName, childTree, true);
+        }
       }
+      return parent;
+    } catch (Exception ioe) {
```
Comment (AymaneS) [inline src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java:119]: Could we have a more precise exception type?

Comment (roxanedelpeyratactiveviam) [reply src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java]: yes, it is supposed to be an IOException

**Review by AymaneS — CHANGES_REQUESTED** [review]
Looks good to me

**Review by AymaneS — DISMISSED** [review]
Looks good to me

File: src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java [hunk 2]
```diff
@@ -51,27 +50,28 @@
...
-    snapshotter.eraseAndImport(
-        folderName + Paths.SEPARATOR + Tree.DASHBOARDS, loadDirectory(Paths.DASHBOARDS));
-    snapshotter.eraseAndImport(
-        folderName + Paths.SEPARATOR + Tree.WIDGETS, loadDirectory(Paths.WIDGETS));
+    try {
+      InputStream res =
+          dashboardTreeResolver.getClassLoader().getResourceAsStream(Paths.INITIAL_CONTENT);
+      snapshotter.eraseAndImport(ContentServerConstants.Paths.UI, res);
+    } catch (Exception e) {
+      LOGGER.error("Cannot load the initial content file");
```
Comment (mdellabani) [inline src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java:71]: shouldn't we rather throw here ?

File: src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java [hunk 3]
```diff
@@ -109,44 +93,49 @@ private static File getSubDirectory(final String folderName) {
...
-  private static JsonNode loadFileIntoNode(File file) {
+  private static JsonNode loadFileIntoNode(InputStream inputStream) {
     JsonNode loadedFile = JsonNodeFactory.instance.objectNode();
     try {
       final ObjectMapper mapper = new ObjectMapper();
-      loadedFile = mapper.readTree(new FileInputStream(file));
+      loadedFile = mapper.readTree(inputStream);
     } catch (Exception e) {
-      LOGGER.warn("Unable to find file " + file.getName());
+      LOGGER.warn("Unable to read the json file");
```
Comment (mdellabani) [inline src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java:138]:
```suggestion
      LOGGER.warn("Unable to map read the input stream as Json");
```

File: src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java [hunk 4]
```diff
@@ -109,44 +93,49 @@ private static File getSubDirectory(final String folderName) {
...
+          final SnapshotContentTree childNode = createEmptyDirectoryNode();
+          SnapshotContentTree childTree =
+              createDirectoryTree(root + Paths.SEPARATOR + childName, childNode);
+          parent.putChild(childName, childTree, true);
+        }
       }
+      return parent;
+    } catch (IOException ioe) {
+      LOGGER.error("Unable to retrieve directory {} from resources. The import will fail.", root);
+      return null;
```
Comment (mdellabani) [inline src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java:122]: same shouldn't we throw ?

**Review by mdellabani — CHANGES_REQUESTED** [review]
Almost good.
I think we should throw a message if we are not able to load the resources. At least we throw a runtime exception so that the user sees it clearly and take manual action

**Review by mdellabani — APPROVED** [review]

**Pull request merged**